src/main/scala/replpp/package.scala

```diff
@@ -27,17 +26,20 @@ package object replpp {
   }
   
   def allPredefCode(config: Config): String = {
-    val lines = config.predefCode.getOrElse("") +: readPredefFiles(config.predefFiles)
-    lines.mkString(lineSeparator)
+    val predefLines: Seq[String] = config.predefCode.getOrElse("") +: readAdditionalFiles(config.predefFiles)
+    val scriptLines = config.scriptFile.map(os.read.lines(_)).getOrElse(Seq.empty)
+    val importedFiles = UsingDirectives.findImportedFilesRecursively(predefLines ++ scriptLines)
+    val linesViaUsingDirective = readAdditionalFiles(importedFiles)
+    (predefLines ++ linesViaUsingDirective).mkString(lineSeparator)
```

Comment:
I'd have expected we have to do everything in reverse order here, i.e. dependencies get evaluated before dependents.

I guess it usually doesn't matter because everything is concatenated here and scala can deal with forward references. But when included files need to do some initialization work, that's harder than it should be right now.